CRM/Case/Form/Task/Batch.php

```diff
@@ -0,0 +1,122 @@
+ +--------------------------------------------------------------------+
+ | This file is a part of CiviCRM.                                    |
+ |                                                                    |
+ | CiviCRM is free software; you can copy, modify, and distribute it  |
+ | under the terms of the GNU Affero General Public License           |
+ | Version 3, 19 November 2007 and the CiviCRM Licensing Exception.   |
+ |                                                                    |
+ | CiviCRM is distributed in the hope that it will be useful, but     |
+ | WITHOUT ANY WARRANTY; without even the implied warranty of         |
+ | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+ | See the GNU Affero General Public License for more details.        |
+ |                                                                    |
+ | You should have received a copy of the GNU Affero General Public   |
+ | License and the CiviCRM Licensing Exception along                  |
+ | with this program; if not, contact CiviCRM LLC                     |
+ | at info[AT]civicrm[DOT]org. If you have questions about the        |
+ | GNU Affero General Public License or the licensing of CiviCRM,     |
+ | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ * @package CRM
+ * @copyright CiviCRM LLC (c) 2004-2017
+ */
+
+/**
+ * This class provides the functionality for batch profile update for cases
+ */
+class CRM_Case_Form_Task_Batch extends CRM_Core_Form_Task_Batch {
+
+  // Must be set to entity table name (eg. civicrm_participant) by child class
+  static $tableName = 'civicrm_case';
+  // Must be set to entity shortname (eg. event)
+  static $entityShortname = 'case';
+
+  /**
+   * Set default values for the form.
+   *
+   * @return array $defaults
+   */
+  public function setDefaultValues() {
+    return parent::setDefaultValues();
+  }
+
+  /**
+   * Process the form after the input has been submitted and validated.
+   *
+   * @return void
+   */
+  public function postProcess() {
+    $params = $this->exportValues();
+
+    if (!isset($params['field'])) {
+      CRM_Core_Session::setStatus(ts("No updates have been saved."), ts('Not Saved'), 'alert');
+      return;
+    }
+
+    $customFields = array();
+    $dateFields = array(
+      'case_created_date',
+      'case_start_date',
+      'case_end_date',
+      'case_modified_date',
+    );
+    foreach ($params['field'] as $key => $value) {
+      $value['id'] = $key;
+
+      if (!empty($value['case_type'])) {
+        $caseTypeId = $value['case_type_id'] = $value['case_type'][1];
+      }
+      unset($value['case_type']);
+
+      //Get the case status
+      $value['status_id'] = (CRM_Utils_Array::value('case_status', $value)) ? $value['case_status'] : CRM_Core_DAO::getFieldValue('CRM_Case_DAO_Case', $key, 'status_id');
```

Comment:
I think there is some personal preference involved there - depending how you edit things longer or shorter likes can be more readable

Reply:
Yep. But happy to clean up code where possible, just limited as to how much time I can spend on clean up at this stage - some of this could be done as smaller commits once we've made this (big) change.  @mickadoo Thanks for the code snippet, I've added it to this PR.